zebrad/tests/acceptance.rs

```diff
@@ -1455,7 +1455,69 @@ async fn tracing_endpoint() -> Result<()> {
     Ok(())
 }
 
-// TODO: RPC endpoint and port conflict tests (#3165)
+#[tokio::test]
+async fn rpc_endpoint() -> Result<()> {
+    use hyper::{body::to_bytes, Body, Client, Method, Request};
+
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let endpoint = format!("127.0.0.1:{}", port);
+    let url = format!("http://{}", endpoint);
+
+    // Write a configuration that has RPC listen_addr set
+    let mut config = default_test_config()?;
+    config.rpc.listen_addr = Some(endpoint.parse().unwrap());
+
+    let dir = testdir()?.with_config(&mut config)?;
+    let child = dir.spawn_child(&["start"])?;
+
+    // Run `zebrad` for a few seconds before testing the endpoint
+    // Since we're an async function, we have to use a sleep future, not thread sleep.
+    tokio::time::sleep(LAUNCH_DELAY).await;
+
+    // Create an http client
+    let client = Client::new();
+
+    // Create a request to call `getinfo` RPC method
+    let req = Request::builder()
+        .method(Method::POST)
+        .uri(url)
+        .header("content-type", "application/json")
+        .body(Body::from(
+            r#"{"jsonrpc": "2.0", "method": "getinfo", "id":123}"#,
```

Comment:
Test Coverage: `lightwalletd` uses JSON-RPC 1.0, please send a request that matches its request format.

(I'll edit this comment to include an example.)

Reply:
Here is the first RPC that `lightwalletd` sends:
```json
{"jsonrpc":"1.0","method":"getblockchaininfo","params":[],"id":1}
```

I got this RPC using the following commands:
```sh
nc -l 127.0.0.1 28232 &
cat << EOF > lightwalletd-zcash.conf
rpcbind=127.0.0.1
rpcport=28232
EOF
lightwalletd --no-tls-very-insecure --zcash-conf-path lightwalletd-zcash.conf --log-file /dev/stdout --data-dir $(mktemp -d)
```

Reply:
I don't think we have to test exactly what lightwalletd does here. We are testing that the rpc port is open and that is ready for any rpc call. That should be it on my opinion here. There are other tests for lightwalletd behaviour.

Reply:
Our goal is to be compatible with `lightwalletd`.

But Gustavo thinks it might be a while before we're actually testing with `lightwalletd`. And the lightwalletd test only runs if we specifically ask for it.

So we will get much better test coverage if we use the exact format that `lightwalletd` sends here. (This is what we do for the metrics and tracing endpoint tests.)

Reply:
I humbly don't agree but i am not going to discuss it. You are suggesting changing `getinfo` method into `getblockchaininfo` ? Because right now `getblockchaininfo` does not exist.

Reply:
I'm fine with either method, both exist in the `main` branch.

See my suggestion at https://github.com/ZcashFoundation/zebra/pull/3624#discussion_r814343961

Reply:
It is not the same at this moment because `getblockchaininfo` does not return `Zebra` string in the response. In regards to the format, yea, i finally agree so i added the suggestion at https://github.com/ZcashFoundation/zebra/pull/3624/commits/d35323b9bda30f5ea5979afe3aac922ea1cc28b0